enarx-prstate

```diff
@@ -25,19 +26,31 @@ def handle(issue, pr, responsible):
     if needed:
         pr.add_to_assignees(*needed)
 
-# If there are any changes requested, the author is responsible.
+# Look through all PRs that have a changes requested review present.
 for issue in github.search_issues(f"org:enarx is:pr is:public is:open -is:draft review:changes_requested"):
     pr = issue.as_pull_request()
 
-    handle(issue, pr, {pr.user.login})
-
-# Otherwise, requested reviewers and reviewers with dismissed reviews are.
-# This handles both where the author requests re-review and where, after
-# approvals, reviews were dismissed due to a new push.
-for issue in github.search_issues(f"org:enarx is:pr is:public is:open -is:draft -review:changes_requested"):
-    pr = issue.as_pull_request()
-
-    dismissed = {r.user.login for r in pr.get_reviews() if r.state == 'DISMISSED'}
-    requested = {r.login for r in pr.get_review_requests()[0]}
-
-    handle(issue, pr, dismissed | requested)
+    # Collect reviewers whose reviews have been addressed in a set.
+    addressed_reviewers = set()
+
+    # Get the latest review for individual reviewers.
+    latest_reviews = {i.user.id: i for i in pr.get_reviews() if i.state == "CHANGES_REQUESTED"}
```

Comment:
I think the tradition when using single letter variable names is to use either:
1. the first letter of the type (so `r` for reviews)
2. the letter `x` (or `x` and `y` for nested iterations)

Reply:
Fixed in my latest local revision.